### ISSUE TITLE
I can't visualize Class_loss and Mask_loss in Tensorboard

### ISSUE THREAD

**Issue by ahasan016** [matterport/Mask_RCNN#2162]
I am training maskrcnn-matterport with my custom dataset and while training I can only visualize loss and val_loss. 
![tensorboard](https://user-images.githubusercontent.com/11074869/81086843-4f7fd500-8ef9-11ea-8a69-c039a32d2ef8.PNG)
But I also want to visualize class loss, mask loss etc. like 
![detection_tensorboard](https://user-images.githubusercontent.com/11074869/81087002-7ccc8300-8ef9-11ea-968f-7f28765a5263.png)

I am simply using "tensorboard --logdir=<log_dir>" command to visualize. 
hope someone can help me.. Thanks

**Comment by FTRobles** [comment]
Hi, I am facing the same problem and I haven´t found an answer

**Comment by Jonathan-Gore** [comment]
ahsan, could you briefly describe or paste some of the code you're using to visualize this in tensorboard? Because I can't get anything aside from the model structure visualized in tensorboard, it just says that the data isn't in the log file.

**Comment by ahasan016** [comment]
> ahsan, could you briefly describe or paste some of the code you're using to visualize this in tensorboard? Because I can't get anything aside from the model structure visualized in tensorboard, it just says that the data isn't in the log file.

Hi Jonathan, 

My main function looks like this : 

     if __name__ == '__main__':

    # Validate arguments
    from tensorflow.python.client import device_lib
    print(device_lib.list_local_devices())
   


    config = ICENUMConfig()
    config.display()
    #DEVICE = "/gpu:0"
  
    model = modellib.MaskRCNN(mode="training", config=config, model_dir=DEFAULT_LOGS_DIR)
                                 
    model.keras_model.metrics_tensors = []
    weights_path = COCO_WEIGHTS_PATH
    # Download weights file
    model.load_weights(weights_path, by_name=True, exclude=[
             "mrcnn_class_logits", "mrcnn_bbox_fc",
             "mrcnn_bbox", "mrcnn_mask"])

    train(model) 

And the "train" function : 

    def train(model):
   
    # Training dataset.
    dataset_train = FoodDataset()
   
    dataset_train.load_food(ROOT_DIR,"Train_Numbers1+2+3+4")
    dataset_train.prepare()

    #Validation dataset
    dataset_val = FoodDataset()
 
    dataset_val.load_food(ROOT_DIR,"VAL_Numbers1+2+3")
    dataset_val.prepare()

    print("Training network heads")
    model.train(dataset_train,dataset_val,
                learning_rate=config.LEARNING_RATE,
                epochs=300,
                layers='all')

Model.train leads to : 

        def train(self, train_dataset, val_dataset, learning_rate, epochs, layers,
              augmentation=None, custom_callbacks=None, no_augmentation_sources=None):
        
        assert self.mode == "training", "Create model in training mode."

        # Pre-defined layer regular expressions
        layer_regex = {
            # all layers but the backbone
            "heads": r"(mrcnn\_.*)|(rpn\_.*)|(fpn\_.*)",
            # From a specific Resnet stage and up
            "3+": r"(res3.*)|(bn3.*)|(res4.*)|(bn4.*)|(res5.*)|(bn5.*)|(mrcnn\_.*)|(rpn\_.*)|(fpn\_.*)",
            "4+": r"(res4.*)|(bn4.*)|(res5.*)|(bn5.*)|(mrcnn\_.*)|(rpn\_.*)|(fpn\_.*)",
            "5+": r"(res5.*)|(bn5.*)|(mrcnn\_.*)|(rpn\_.*)|(fpn\_.*)",
            # All layers
            "all": ".*",
        }
        if layers in layer_regex.keys():
            layers = layer_regex[layers]

        # Data generators
        train_generator = data_generator(train_dataset, self.config, shuffle=True,
                                         augmentation=augmentation,
                                         batch_size=self.config.BATCH_SIZE,
                                         no_augmentation_sources=no_augmentation_sources)
        val_generator = data_generator(val_dataset, self.config, shuffle=True,
                                       batch_size=self.config.BATCH_SIZE)

        # Callbacks
        callbacks = [
            keras.callbacks.TensorBoard(log_dir=self.log_dir,
                                        histogram_freq=0, write_graph=True, write_images=False),
            keras.callbacks.ModelCheckpoint(self.checkpoint_path,
                                            verbose=0, save_weights_only=True),
        ]
	
        # Add custom callbacks to the list
        if custom_callbacks:
            callbacks += custom_callbacks

        # Train
        log("\nStarting at epoch {}. LR={}\n".format(self.epoch, learning_rate))
        log("Checkpoint Path: {}".format(self.checkpoint_path))
        self.set_trainable(layers)
        self.compile(learning_rate, self.config.LEARNING_MOMENTUM)

        # Work-around for Windows: Keras fails on Windows when using
        # multiprocessing workers. See discussion here:
        # https://github.com/matterport/Mask_RCNN/issues/13#issuecomment-353124009
        if os.name == 'nt':
            workers = 0
        else:
            workers = multiprocessing.cpu_count()

        self.keras_model.fit_generator(
            train_generator,
            initial_epoch=self.epoch,
            epochs=epochs,
            steps_per_epoch=self.config.STEPS_PER_EPOCH,
            callbacks=callbacks,
            validation_data=val_generator,
            validation_steps=self.config.VALIDATION_STEPS,
            max_queue_size=100,
            workers=workers,
            use_multiprocessing=True,
        )
        #print("abc")
        self.epoch = max(self.epoch, epochs) 

for visualization , I simply run "tensorboard --logdir=<log_dir>" command on anacondra prompt.

**Comment by ahasan016** [comment]
> Hi, I am facing the same problem and I haven´t found an answer

any solution yet?

**Comment by andreap9** [comment]
>  Check if you are excluding layers when loading the weights for detection. I think that is the problem.

**Comment by Tonks684** [comment]
> > Check if you are excluding layers when loading the weights for detection. I think that is the problem.

Did this provide a solution?

**Comment by ahasan016** [comment]
Not really. I didn't even understand what he meant. I tried to visualize while training not during detection.

**Comment by Tonks684** [comment]
> Not really. I didn't even understand what he meant. I tried to visualize while training not during detection.

I see were you able to visualise accuracy by epoch?

**Comment by ahasan016** [comment]
I only could visualize the total training and total validation loss graphs. Not individual graphs.

**Comment by mummpl** [comment]
Hey guys,
I solved this by downgrading keras to 2.0.8
The tf version is probably 1.3 to 1.5 (using it in colab which only allows me to switch between some version of tf 1 and 2). 
Hope this helps you!

**Comment by AndreyStille** [comment]
> Hey guys,
> I solved this by downgrading keras to 2.0.8
> The tf version is probably 1.3 to 1.5 (using it in colab which only allows me to switch between some version of tf 1 and 2).
> Hope this helps you!

Hi I have same issue. But cannot deal with keras in colab. I reinstall it and when I make (print(tf.keras.__version__)) it still returns (2.2.4-tf). How did you downgrade keras in colab?

**Comment by mummpl** [comment]
> > Hey guys,
> > I solved this by downgrading keras to 2.0.8
> > The tf version is probably 1.3 to 1.5 (using it in colab which only allows me to switch between some version of tf 1 and 2).
> > Hope this helps you!
> 
> Hi I have same issue. But cannot deal with keras in colab. I reinstall it and when I make (print(tf.keras.**version**)) it still returns (2.2.4-tf). How did you downgrade keras in colab?

pip install -I keras==2.0.8 should do the trick

**Comment by AndreyStille** [comment]
> > > Hey guys,
> > > I solved this by downgrading keras to 2.0.8
> > > The tf version is probably 1.3 to 1.5 (using it in colab which only allows me to switch between some version of tf 1 and 2).
> > > Hope this helps you!
> > 
> > 
> > Hi I have same issue. But cannot deal with keras in colab. I reinstall it and when I make (print(tf.keras.**version**)) it still returns (2.2.4-tf). How did you downgrade keras in colab?
> 
> pip install -I keras==2.0.8 should do the trick
Thanks for response
But I still cant deal with it. 
I install keras==2.0.8 succesfully and restart my runtime.
Then there are two ways:
1) I make  %tensorflow_version 1.x . Then import keras. But that way the version of keras is still 2.3.1.
2) I make import keras first. Then I can't make %tensorflow_version 1.x and it uses tf version 2.4.1 which is inpropriate.
